src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,74 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  std::shared_ptr<current_namest> global_names;
+
+  /// Ages the generation of an ssa expression by one.
+  void increase_generation(const irep_idt l1_identifier, const ssa_exprt &lhs)
+  {
+    current_names.emplace(l1_identifier, std::make_pair(lhs, 0));
+    if(global_names != nullptr)
+      global_names->emplace(l1_identifier, std::make_pair(lhs, 0));
+
+    increase_generation(l1_identifier);
+  }
+
+  /// Ages the generation of an ssa expression by one. Does nothing
+  /// if there isn't an expression keyed by the l1 identifier.
+  void increase_generation(const irep_idt identifier)
+  {
+    // If we can't find the name in the local scope, don't incr ease the global
+    // even if it exists there.
+    auto current_names_iter = current_names.find(identifier);
```

Comment:
It it always a bug if symex tries to allocate a new generation of something that hasn't been declared?

Reply:
Question still outstanding

Reply:
I'm not sure, but for me the question is whether highlighting that lack here is actually useful. If I did put any sort of assertion in I'd just make this private and add another public method like increase_generation_if_exists instead, so nothing would really change.